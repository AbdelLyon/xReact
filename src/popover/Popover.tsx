import { PropsWithChildren, ReactNode } from "react";

import {
  Popover as PopoverRoot,
  PopoverContent,
  PopoverTrigger,
  PopoverClose,
} from "@/ui/popover";
import { cn } from "@/utils";

type Props = PropsWithChildren<{
  classNames?: {
    popover?: string;
    trigger?: string;
  };
  trigger: ReactNode;
}>;

const Popover = ({
  classNames,
  trigger,

  children,
}: Props) => {
  return (
    <PopoverRoot>
      <PopoverTrigger
        asChild
        className={cn("border border-border rounded-md", classNames?.trigger)}
      >
        <div>{trigger}</div>
      </PopoverTrigger>
      <PopoverContent className={cn("mt-1 z-50", classNames?.popover)}>
        {children}
      </PopoverContent>
    </PopoverRoot>
  );
};

export { Popover, PopoverClose };
