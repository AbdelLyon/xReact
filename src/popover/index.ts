import "@/index.css";
export { Popover, PopoverClose } from "@/popover/Popover";
