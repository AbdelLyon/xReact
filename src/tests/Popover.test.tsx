import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Popover } from "@/popover/Popover";

describe("Popover", () => {
  it("matches the snapshot", () => {
    const triggerText = "Click me";
    const popoverContent = "Popover content";
    const { asFragment } = render(
      <Popover
        classNames={{ trigger: "custom-trigger", popover: "custom-popover" }}
        trigger={<button>{triggerText}</button>}
      >
        <div>{popoverContent}</div>
      </Popover>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
