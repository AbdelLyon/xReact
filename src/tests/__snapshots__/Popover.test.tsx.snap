// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="border border-border rounded-md custom-trigger"
    data-state="closed"
    type="button"
  >
    <button>
      Click me
    </button>
  </div>
</DocumentFragment>
`;
